waggle-dance-core/src/test/java/com/hotels/bdp/waggledance/client/compatibility/HiveCompatibleThriftHiveMetastoreIfaceFactoryTest.java

```diff
@@ -98,6 +98,37 @@ public void underlyingExceptionIsThrownWhenCompatibilityFails() throws Exception
     }
   }
 
+  @Test
+  public void compatibilityExceptionIsThrownWhenCompatibilityFailsOnTException() throws Exception {
+    CloseableThriftHiveMetastoreIface thriftHiveMetastoreIface = factory.newInstance(delegate);
+    GetTableRequest tableRequest = new GetTableRequest(DB_NAME, TABLE_NAME);
+    when(delegate.get_table_req(tableRequest))
+        .thenThrow(new TApplicationException("ApplicationException, should not be thrown"));
+    when(delegate.get_table(DB_NAME, TABLE_NAME))
+        .thenThrow(new NoSuchObjectException("Should be thrown, this is called from compatiblity layer"));
+    try {
+      thriftHiveMetastoreIface.get_table_req(tableRequest);
+      fail("exception should have been thrown");
+    } catch (NoSuchObjectException e) {
+      assertThat(e.getMessage(), is("Should be thrown, this is called from compatiblity layer"));
```

Comment:
Minor, but you could keep a reference to the exception object above and check object equality rather than the message.